tests/trees/test_regression_hoeffding_adaptive_tree.py

```diff
@@ -65,38 +63,48 @@ def test_hoeffding_tree_perceptron():
 
     learner = HoeffdingAdaptiveTreeRegressor(leaf_prediction='perceptron', random_state=1)
 
-    cnt = 0
     max_samples = 500
     y_pred = array('d')
     y_true = array('d')
     wait_samples = 10
 
-    while cnt < max_samples:
+    for cnt in range(max_samples):
         X, y = stream.next_sample()
         # Test every n samples
         if (cnt % wait_samples == 0) and (cnt != 0):
             y_pred.append(learner.predict(X)[0])
             y_true.append(y[0])
         learner.partial_fit(X, y)
-        cnt += 1
-
-    expected_predictions = array('d', [1198.4326121743168, 456.36607750881586, 927.9912160545144, 1160.4797981899128,
-                                       506.50541829176535, -687.8187227095925, -677.8120094065415, 231.14888704761225,
-                                       -284.46324039942937, -255.69195985557175, 47.58787439365423, -135.22494016284043,
-                                       -10.351457437330152, 164.95903200643997, 360.72854984472383, 193.30633911830088,
-                                       -64.23638301570358, 587.9771578214296, 649.8395655757931, 481.01214222804026,
-                                       305.4402728117724, 266.2096493865043, -445.11447171009775, -567.5748694154349,
-                                       -68.70070048021438, -446.79910655850153, -115.892348067663, -98.26862866231015,
-                                       71.04707905920286, -10.239274802165584, 18.748731569441812, 4.971217265129857,
-                                       172.2223575990573, -655.2864976783711, -129.69921313686626, -114.01187375876822,
-                                       -405.66166686550963, -215.1264381928009, -345.91020370426247, -80.49330468453074,
-                                       108.78958382083302, 134.95267043280126, -398.5273538477553, -157.1784910649728,
-                                       219.72541225645654, -100.91598162899217, 80.9768574308987, -296.8856956382453,
-                                       251.9332271253148])
+
```

Comment:
Why are results different. This is also reflected in the MAE. Changing the operators should not affect the results.

Reply:
Seeing the predictions, it looks like the computing precision is improved (differences are smaller than about 1e-05) so I assume it's either an earlier cast or one to a type with better precision.
But I don't know why that happens...

Reply:
@jacobmontiel I've been thinking about this & I wondered wether it truly mattered: if it's just a matter of casting, could we not simply switch to the much more readable infix notation, regardless of the computation difference?

Reply:
@jacobmontiel this was due to `+=`, probably because it's inplace.

Reply:
Did the results reverted back to their previous form since you changed the operators? That's unfortunate that the syntax sugar is not treated as a `ufunc`.